File: OOP/3_private_encaptulaiton.cpp
#include <iostream>
#include <string>
using namespace std;

class Status
{
private:
    string status; // Private variable
    // Private method
    void married()
    {
        cout << "This is a private method" << endl;
    }

public:
    string name; // Public variable

    // Constructor
    Status()
    {
        status = "Single";
        name = "Arman";
    }

    string getStatus()
    {
        return status;
    }

    void getMarried()
    {
        married(); // Call private method
    }
    string setStatus(string newStatus)
    {
        status = newStatus;
        return "done";
    }
};

class New : public Status
{
public:
    // Constructor
    New() : Status() {}

    string getStatus()
    {
        return Status::getStatus();
    }
};

int main()
{

    Status myobj;
    New newObj;

    cout << newObj.setStatus("mm") << endl;
    cout << myobj.getStatus() << endl;

    myobj.getMarried();
    cout << "Name: " << myobj.name << ", Status: " << myobj.getStatus() << endl;

    myobj.setStatus("Engaged");
    cout << myobj.getStatus() << endl;

    return 0;
}

File: OOP/3_private_encaptulaiton.py
class Status:
    def __init__(self) -> None:
        self.__status = 'Single' #eta private var jeta baire teke directly access kora jabenaa
        self.name = 'Arman' #eta k abr access kora jabe
    def __married(self):
        print('eta private method')
    def get_status(self):
        return self.__status #private var er value deka jabe
    def get_married(self):
        return self.__married() #private method access korar jnno
    def set_status(self,status):
        self.__status = status #eta diye private variable set korte parbo
        return 'done'

class New(Status):
    def __init__(self) -> None:
        super().__init__()
    def get_status(self):
        return super().get_status()

myobj = Status()
newObj = New()
print(newObj.set_status('mm'))
# print(myobj.__status) eta kaj korbena
print(myobj.get_status()) 
print(myobj.get_married())
print(myobj.__dict__)
myobj.set_status('Engaged')
print(myobj.get_status())